### ISSUE TITLE
IDs not replaced in links

### ISSUE THREAD

**Issue by waruyama** [iconfu/svg-inject#21]
Elements can link to another element with the xlink:href or href attribute. The IDs in these links are currently not replaced by the new IDs. Here is an example how this can look in SVG:

```
  <linearGradient id="gradient1" href="#gradient2" />
  <linearGradient id="gradient2" xlink:href="#gradient3" />
  <linearGradient id="gradient3" gradientUnits="userSpaceOnUse" x1="0" y1="800" x2="0" y2="800">
   <stop offset="0" style="stop-color:#FFFF00"/>
   <stop offset="1" style="stop-color:#00FF00"/>
  </linearGradient>
```
In this example a link to `#gradient1` would actually use `#gradient3`

**Comment by waruyama** [comment]
Since links are only permitted to an element of the same type (for example a link in a 'linearGradient' may only reference another 'linearGradient'), only IDs in links inside IRI referenceable elements must be replaced.

For IRI referenceable elements the IDs in links can be replaced like this:

```javascript
      ['xlink:href', 'href'].forEach(function(refAttrName) {
        var iri = idElem.getAttribute(refAttrName);
        if (/^\s*#/.test(iri)) { // Check if iri is non-null and has correct format
          idElem.setAttribute(refAttrName, iri.trim() + idSuffix)
        }
      });
```

**Comment by iconexperience** [comment]
Fixed in a809111

**Comment by waruyama** [comment]
Fixed in a809111
